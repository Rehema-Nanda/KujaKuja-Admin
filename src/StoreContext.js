// https://mobx-react.js.org/recipes-context

import React from "react";
import PropTypes from "prop-types";
import RootStore from "./stores/RootStore";

export const StoreContext = React.createContext(null);

export const StoreProvider = ({ children }) => {
    const store = new RootStore();
    return <StoreContext.Provider value={store}>{children}</StoreContext.Provider>;
};
StoreProvider.propTypes = {
    children: PropTypes.node.isRequired,
};

export const StoreConsumer = StoreContext.Consumer;

export const useStore = () => {
    const store = React.useContext(StoreContext);
    if (!store) {
        // this is especially useful in TypeScript so you don't need to be checking for null all the time
        throw new Error("useStore must be used within a StoreProvider.");
    }
    return store;
};
