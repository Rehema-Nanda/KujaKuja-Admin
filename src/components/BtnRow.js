import React from "react";
import PropTypes from "prop-types";
import { Row, Col, Button } from "reactstrap";
import "../index.css";
import { observer } from "mobx-react";
import { StoreContext } from "../StoreContext";

class BtnRow extends React.PureComponent {
    render() {
        const { userStore: { userInfo } } = this.context;
        const { buttonLabel, cancelForm } = this.props;

        return (

            <Row className="btn-row">
                <Col>
                    {
                        userInfo
                        && userInfo.length > 0
                        && userInfo[0].isAdmin
                        && buttonLabel
                        && <Button color="primary" className="primary">{buttonLabel}</Button>
                    }
                </Col>
                <Col>
                    <Button color="primary" className="primary cancel" onClick={cancelForm}>Back</Button>
                </Col>
            </Row>

        );
    }
}

BtnRow.propTypes = {
    buttonLabel: PropTypes.string.isRequired,
    cancelForm: PropTypes.func.isRequired,
};

BtnRow.contextType = StoreContext;

export default observer(BtnRow);
