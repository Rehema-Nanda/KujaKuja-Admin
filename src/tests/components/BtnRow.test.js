import React from "react";
import { mount } from "enzyme";
import BtnRow from "../../components/BtnRow";
import { StoreProvider } from "../../StoreContext";

describe("BtnRow component", () => {
    const testProps = {
        buttonLabel: "TestLabel",
        cancelForm: jest.fn(),
    };
    const wrapper = mount(<BtnRow {...testProps} />, { wrappingComponent: StoreProvider }); // eslint-disable-line react/jsx-props-no-spreading

    it("renders correctly", () => {
        expect(wrapper.debug()).toMatchSnapshot();
    });
});
