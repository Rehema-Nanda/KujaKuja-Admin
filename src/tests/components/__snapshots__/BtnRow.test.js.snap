// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BtnRow component renders correctly 1`] = `
"<BtnRow buttonLabel=\\"TestLabel\\" cancelForm={[Function: mockConstructor]}>
  <Row className=\\"btn-row\\" tag=\\"div\\">
    <div className=\\"btn-row row\\">
      <Col tag=\\"div\\" widths={{...}}>
        <div className=\\"col\\" />
      </Col>
      <Col tag=\\"div\\" widths={{...}}>
        <div className=\\"col\\">
          <Button color=\\"primary\\" className=\\"primary cancel\\" onClick={[Function: mockConstructor]} tag=\\"button\\">
            <button type=\\"button\\" onClick={[Function: bound onClick]} className=\\"primary cancel btn btn-primary\\" aria-label={{...}}>
              Back
            </button>
          </Button>
        </div>
      </Col>
    </div>
  </Row>
</BtnRow>"
`;
